client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/TransactionalRemoteCacheImpl.java

```diff
@@ -42,7 +42,8 @@
    private final TransactionManager transactionManager;
    private final TransactionTable transactionTable;
 
-   private final Function<K, MetadataValue<V>> remoteGet = super::getWithMetadata;
+   // TODO: the remote get is force to be sync and is blocking!
```

Comment:
create a JIRA please!!!

Reply:
I created https://issues.redhat.com/browse/ISPN-11633 a while back. Let me add it as a comment here though.